src/main/java/cms/gov/madie/terminology/controller/VsacController.java

```diff
@@ -38,28 +39,35 @@ public ResponseEntity<String> getValueSet(
       @RequestParam(required = false, name = "profile") String profile,
       @RequestParam(required = false, name = "includeDraft") String includeDraft,
       @RequestParam(required = false, name = "release") String release,
-      @RequestParam(required = false, name = "version") String version)
-      throws InterruptedException, ExecutionException {
+      @RequestParam(required = false, name = "version") String version) {
 
     log.debug("Entering: getValueSet()");
 
-    String serviceTicket = vsacService.getServiceTicket(tgt);
-
     RetrieveMultipleValueSetsResponse valuesetResponse =
-        vsacService.getValueSet(oid, serviceTicket, profile, includeDraft, release, version);
+        vsacService.getValueSet(oid, tgt, profile, includeDraft, release, version);
 
-    ValueSet fhirValueSet = vsacService.convertToFHIRValueSet(valuesetResponse, oid);
+    ValueSet fhirValueSet = vsacService.convertToFHIRValueSet(valuesetResponse);
     log.debug("valueset id = " + fhirValueSet.getId());
 
     String serialized = serializeFhirValueset(fhirValueSet);
 
     return ResponseEntity.ok().body(serialized);
   }
 
-  protected String serializeFhirValueset(ValueSet fhirValueSet) {
-
-    String serialized = parser.encodeResourceToString(fhirValueSet);
+  @PutMapping(path = "/value-sets/searches", produces = MediaType.APPLICATION_JSON_VALUE)
```

Comment:
@adongare It's nice work Ashok! I'm not sure if this is supposed to replace the current work of getting value set one by one from front end, if it is, I think the ideal solution would be handling the failed value sets and get them mapped with elm standard errors so front end can get both success value sets as well as ready to use failed ones.

Reply:
Thanks @sb-cecilialiu! We can modify this lil bit to replace current work but I'm not fully aware of that requirements. I can work with you to understand it a little bit more and draft s story or work on it.